rover/src/owr_drive_controls/src/swerveDrive.cpp

```diff
@@ -16,96 +16,54 @@
-            pow(velMsg->linear.y, 2))) / rotationRadius;
-        ROS_INFO(
-            "turnAngle %lf, rotationRadius %lf, rotationCenter  %lf, %lf, %lf",
-            turnAngle, rotationRadius, rotationCentre.x, rotationCentre.y, rotationCentre.z);
-        // calculate the radiuses of each wheel about the rotation center
-        // NOTE: if necisary this could be optimised
+        rotationCentre.y = sqrt(pow(rotationRadius,2)-pow(HALF_ROVER_WIDTH_X,2));
+        //magnitude velocity over rotation radius
+        //const double angularVelocity = (velMsg->linear.y / cos(turnAngle)) / rotationRadius;
+        const double angularVelocity = fabs(sqrt(pow(velMsg->linear.x, 2) + pow(velMsg->linear.y, 2))) / rotationRadius;
+        ROS_INFO("turnAngle %lf, rotationRadius %lf, rotationCenter  %lf, %lf, %lf",turnAngle, rotationRadius, rotationCentre.x, rotationCentre.y, rotationCentre.z);
+        //calculate the radiuses of each wheel about the rotation center
+        //NOTE: if necisary this could be optimised
         double closeBackR = fabs(rotationCentre.y - ROVER_CENTRE_2_WHEEL_Y);
         double farBackR = fabs(rotationCentre.y + ROVER_CENTRE_2_WHEEL_Y);
-        double closeFrontR =
-            sqrt(pow(closeBackR, 2) + pow(FRONT_W_2_BACK_W_X, 2));
-        double farFrontR = sqrt(pow(farBackR, 2) + pow(FRONT_W_2_BACK_W_X, 2));
-
-        // V = wr
+        double closeFrontR = sqrt(pow(closeBackR,2) + pow(FRONT_W_2_BACK_W_X,2));
+        double farFrontR = sqrt(pow(farBackR,2) + pow(FRONT_W_2_BACK_W_X,2));
+        
+        //V = wr
         double closeBackV = closeBackR * angularVelocity;
         double farBackV = farBackR * angularVelocity;
         double closeFrontV = closeFrontR * angularVelocity;
         double farFrontV = farFrontR * angularVelocity;
-
-        // work out the front wheel angles
-        double closeFrontAng = DEG90 - atan2(closeBackR, FRONT_W_2_BACK_W_X);
-        double farFrontAng = DEG90 - atan2(farBackR, FRONT_W_2_BACK_W_X);
-
-        // if we are in reverse,
-        // we just want to go round the same circle in the opposite direction
-        // (I think...)
-        if (velMsg->linear.x < 0) {
-            // flip all the motorVs
+        
+        //work out the front wheel angles
+        double closeFrontAng = DEG90-atan2(closeBackR,FRONT_W_2_BACK_W_X);
+        double farFrontAng = DEG90-atan2(farBackR,FRONT_W_2_BACK_W_X);
+        
+        //if we are in reverse, we just want to go round the same circle in the opposite direction (I think...)
+        if(velMsg->linear.x < 0) {
+            //flip all the motorVs
             closeFrontV *=-1.0;
             farFrontV *=-1.0;
             farBackV *=-1.0;
             closeBackV *=-1.0;
         }
-
-        // work out which side to favour
-        if (0 <= turnAngle && turnAngle <= M_PI) {
-            /* Turn Right */
+        
+        
+        //work out which side to favour
+        if(0 <= turnAngle && turnAngle <= M_PI) {
```

Comment:
Please revert these changes to the original file, want to make sure bluetongue is not broken on m

Reply:
This file has been reverted. Note that this PR is merging into my main topic branch `sajid/RL-11-crab-steer` where I will do more work, not `master`.